dvc/repo/stage.py

```diff
@@ -299,14 +303,16 @@ def collect_granular(
             return [StageInfo(stage) for stage in self.repo.stages]
 
         stages, file, _ = _collect_specific_target(
-            self, target, with_deps, recursive, accept_group
+            self, target, with_deps, recursive, accept_group, glob=glob
```

Comment:
Now that `repro` will support output name as a target, I don't think we should `glob` for the stage names at all. Or, we could think of a different flags to glob stage names.

Reply:
Doing this way, if the glob does match within the `dvc.yaml`, it will not try to collect from other stages. Globbing outputs should be global.